resourcevalidator/prefer_write_only_attribute_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package resourcevalidator_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/resourcevalidator"
+)
+
+func TestPreferWriteOnlyAttribute(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		validators []resource.ConfigValidator
+		req        resource.ValidateConfigRequest
+		expected   *resource.ValidateConfigResponse
+	}{
+		"valid-warning-diag": {
+			validators: []resource.ConfigValidator{
+				resourcevalidator.PreferWriteOnlyAttribute(
+					path.MatchRoot("oldAttribute1"),
+					path.MatchRoot("writeOnlyAttribute1"),
+				),
+				resourcevalidator.PreferWriteOnlyAttribute(
+					path.MatchRoot("oldAttribute2"),
+					path.MatchRoot("writeOnlyAttribute2"),
+				),
+			},
+			req: resource.ValidateConfigRequest{
+				ClientCapabilities: resource.ValidateConfigClientCapabilities{WriteOnlyAttributesAllowed: true},
+				Config: tfsdk.Config{
+					Schema: schema.Schema{
+						Attributes: map[string]schema.Attribute{
+							"oldAttribute1": schema.StringAttribute{
+								Optional: true,
+							},
+							"writeOnlyAttribute1": schema.StringAttribute{
+								Optional: true,
+							},
```

Comment:
nit: I know this doesn't affect the test in particular, but the schema should have WriteOnly here 😄  (same in other tests)